.github/workflows/makefile.yml

```diff
@@ -135,6 +135,31 @@ jobs:
       run: >
         source env/activate && make test
 
+  # Run ESLint for JavaScript code.
+  lint-js:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+    # Check out with additionally pulling the "main" branch, so that the script
+    # can build diff against it.
+    - name: Check out sources
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Create extra checkout branches
+      run: >
+        git branch main origin/main
```

Comment:
should we get a diff with the base branch here instead?

Reply:
we can just put ```git branch main HEAD^1``` here instead

Reply:
Good question. Did I get it correctly you're thinking about the case of CL chains: whether an bug in CL1 should trigger linter errors in all CL1/CL2/etc., or should the linter only complain in CL1?

Currently the PR implements the former, because the linter in each CL looks at the diff from `main`.

If we are to implement the latter, I think we should look at things that Github provides natively, like `github.base_ref`/`GITHUB_BASE_REF`. The `HEAD^1` trick is interesting, but I'm not sure it works correctly in all cases, e.g. when the CL contains multiple merge commits.